Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -1419,44 +1419,83 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Auto Purge Enabled(default) - Dele
         REQUIRE(c4coll_getDocumentCount(coll) == 1);
     }
 
+    // Push parameter
     C4ParamsSetter paramsSetter = [&replCollections](C4ReplicatorParameters& c4Params) {
         c4Params.collectionCount = replCollections.size();
         c4Params.collections = replCollections.data();
     };
+    // Push to the remote
     replicate(paramsSetter);
 
     // Delete the doc and push it:
+    //
     {
         TransactionHelper t(db);
         C4Error error;
-        for (auto doc : docs) {
+        for (auto& doc : docs) {
             doc = c4doc_update(doc, kC4SliceNull, kRevDeleted, ERROR_INFO(error));
             REQUIRE(error.code == 0);
-            REQUIRE(doc);
-            REQUIRE(doc->flags == (C4DocumentFlags)(kDocExists | kDocDeleted));
         }
     }
-    for (auto coll : collections) {
-        CHECK(c4coll_getDocumentCount(coll) == 0);
+    for (size_t i = 0; i < collectionCount; ++i) {
+        REQUIRE(docs[i]);
+        REQUIRE(docs[i]->flags == (C4DocumentFlags)(kDocExists | kDocDeleted));
+        REQUIRE(c4coll_getDocumentCount(collections[i]) == 0);
     }
+    // Push the deleted docs
     replicate(paramsSetter);
 
-    // Apply a pull and verify that the document is not purged.
+    bool deleteThenCreate = true;
+    SECTION("Delete then Create") {
+        // Create a new doc with the same id that was deleted:
+        {
+            TransactionHelper t(db);
+            for (size_t i = 0; i < collectionCount; ++i) {
+                C4Error error;
+                docs[i] = c4coll_createDoc(collections[i], slice(docID),
+                                           json2fleece(bodyJSON.asString().c_str()),
+                                           0, ERROR_INFO(error));
+                REQUIRE(error.code == 0);
+                REQUIRE(docs[i] != nullptr);
+            }
+        }
+        for (auto coll : collections) {
+            REQUIRE(c4coll_getDocumentCount(coll) == 1);
+        }
+    }
+
+    SECTION("Delete without Create") {
```

Comment:
This test is called "delete then create" and it has a section called "Delete without Create".  This seems to just add noise without meaning.

Reply:
I combined two tests in ReplicatorSGTest, "Auto Purge Enabled(default) - Delete Doc" & "Auto Purge Enabled(default) - Delete then Create Doc", into one with two SECTIONs.

Reply:
Then the naming of the test doesn't really match anymore, since it describes only one of the scenarios being tested.

Reply:
updated the name of the test to "Delete Doc or Delete then Create Doc SG"